src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java

```diff
@@ -0,0 +1,264 @@
+
+        if (handlerList.tagCount() == handlers.length){
+            for (int i = 0; i < handlerList.tagCount(); i++)
+            {
+                final NBTTagCompound handlerCompound = handlerList.getCompoundTagAt(i);
+                final IItemHandlerModifiable modifiable = handlers[indexList.getIntAt(i)];
+                if (modifiable instanceof INBTSerializable)
+                {
+                    INBTSerializable serializable = (INBTSerializable) modifiable;
+                    serializable.deserializeNBT(handlerCompound);
+                }
+            }
+        }
+
+        setName(nbt.hasKey(NBT_KEY_NAME) ? nbt.getString(NBT_KEY_NAME) : null);
+    }
+
+    /**
+     * Overrides the stack in the given slot. This method is used by the
+     * standard Forge helper methods and classes. It is not intended for
+     * general use by other mods, and the handler may throw an error if it
+     * is called unexpectedly.
+     *
+     * @param slot  Slot to modify
+     * @param stack ItemStack to set slot to (may be null)
+     * @throws RuntimeException if the handler is called in a way that the handler
+     *                          was not expecting.
+     **/
+    @Override
+    public void setStackInSlot(int slot, final ItemStack stack)
+    {
+        for(IItemHandlerModifiable modifiable : handlers) {
+            if(slot < modifiable.getSlots()) {
+                modifiable.setStackInSlot(slot, stack);
+                return;
+            }
+
+            slot -= modifiable.getSlots();
+        }
+    }
+
+    /**
+     * Returns the number of slots available
+     *
+     * @return The number of slots available
+     **/
+    @Override
+    public int getSlots()
+    {
+        return Arrays.stream(handlers).mapToInt(IItemHandler::getSlots).sum();
+    }
+
+    /**
+     * Returns the ItemStack in a given slot.
+     * <p>
+     * The result's stack size may be greater than the itemstacks max size.
+     * <p>
+     * If the result is null, then the slot is empty.
+     * If the result is not null but the stack size is zero, then it represents
+     * an empty slot that will only accept* a specific itemstack.
+     * <p>
+     * <p/>
+     * IMPORTANT: This ItemStack MUST NOT be modified. This method is not for
+     * altering an inventories contents. Any implementers who are able to detect
+     * modification through this method should throw an exception.
+     * <p/>
+     * SERIOUSLY: DO NOT MODIFY THE RETURNED ITEMSTACK
+     *
+     * @param slot Slot to query
+     * @return ItemStack in given slot. May be null.
+     **/
+    @Override
+    public ItemStack getStackInSlot( int slot)
+    {
+        for(IItemHandlerModifiable modifiable : handlers) {
+            if (slot < modifiable.getSlots()) {
+                return modifiable.getStackInSlot(slot);
+            }
+
+            slot -= modifiable.getSlots();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Introduce a new variable instead of reusing the parameter "slot". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1226)